File: Sample/NendSDK_Sample/NativeAdPageContentViewController.m
//
//  NativeAdPageContentViewController.m
//  NendSDK_Sample
//
//  Copyright (c) 2015年 F@N Communications. All rights reserved.
//

#import "NativeAdPageContentViewController.h"

@interface NativeAdPageContentViewController ()

@property (nonatomic, weak) IBOutlet UIView<NADNativeViewRendering> *adView;
@property (nonatomic, weak) IBOutlet UILabel *label;

@end

@implementation NativeAdPageContentViewController

- (void)setAd:(NADNative *)ad
{
    [ad intoView:self.adView];
}

- (void)setPosition:(NSUInteger)position
{
    self.label.text = [NSString stringWithFormat:@"Page%lu", (unsigned long)position];
}

@end

File: Sample/NendSDK_Swift_Sample/NativeAdPageContentViewController.swift
//
//  NativeAdPageContentViewController.swift
//  NendSDK_Sample
//
//  Copyright © 2015年 F@N Communications. All rights reserved.
//

import UIKit
import NendAd

class NativeAdPageContentViewController: UIViewController {
    
    @IBOutlet fileprivate weak var adView: UIView!
    @IBOutlet fileprivate weak var label: UILabel!
    
    var ad: NADNative! {
        didSet {
            self.ad.intoView(self.adView as! (UIView & NADNativeViewRendering)!, advertisingExplicitly: .promotion)
        }
    }
    
    var position: Int! {
        didSet {
            self.label.text = "Page\(self.position!)"
        }
    }
}
